xbmc/filesystem/CacheStrategy.cpp

```diff
@@ -327,32 +327,32 @@ int64_t CDoubleCache::Seek(int64_t iFilePosition)
   return m_pCache->Seek(iFilePosition); // Normal seek
 }
 
-bool CDoubleCache::Reset(int64_t iSourcePosition, bool clearAnyway)
+bool CDoubleCache::Reset(int64_t iSourcePosition, bool clearAnyway /* = false */)
 {
-  if (!clearAnyway && m_pCache->IsCachedPosition(iSourcePosition)
-      && (!m_pCacheOld || !m_pCacheOld->IsCachedPosition(iSourcePosition)
-          || m_pCache->CachedDataEndPos() >= m_pCacheOld->CachedDataEndPos()))
+  if (clearAnyway || !m_pCache->IsCachedPosition(iSourcePosition))
   {
-    return m_pCache->Reset(iSourcePosition, clearAnyway);
-  }
-  if (!m_pCacheOld)
-  {
-    CCacheStrategy *pCacheNew = m_pCache->CreateNew();
-    if (pCacheNew->Open() != CACHE_RC_OK)
+    // Position not in active cache: swap caches
+    CCacheStrategy* pCacheTmp;
+    if (!m_pCacheOld)
+    {
+      pCacheTmp = m_pCache->CreateNew();
+      if (pCacheTmp->Open() != CACHE_RC_OK)
+      {
```

Comment:
and here

Reply:
Yes, I had the exact same idea until I checked the Open() implementations: It's already being done there.